src/Model/Operation/ProductSyncHandler.php

```diff
@@ -274,6 +286,28 @@ private function handleConfiguratorGroups(SalesChannelProductEntity $product): a
         return $mainProducts;
     }
 
+    private function handleFirstActiveVariant(ProductEntity $product): ProductEntity
+    {
+        $mainProduct = null;
+        $variants = new ProductCollection([$product]);
+
+        foreach ($product->getChildren() as $child) {
+            if ($child === $product->getChildren()->filterByProperty('active', true)->first()) {
+                $mainProduct = $child;
+            } else {
+                $variants->add($child);
+            }
+        }
+
+        if (!$mainProduct) {
+            return $product;
```

Comment:
Do we want that? The function is only called, when the product is inactive. So when no variant is active, I guess we still don't want the inactive main product?